### DISCUSSION TITLE
Unable to use TLS 1.2 from CA

### DISCUSSION THREAD

**Discussion by mjb8034** [wixtoolset/issues#7477] [Questions]
## Bugs

If this issue is a bug:

* Which version of WiX are you building with?

> WiX version in 3.11.2.4516 format

* Which version of Visual Studio are you building with (if any)?

> Visual Studio 2015 Update 3

* Which version of the WiX Toolset Visual Studio Extension are you building with (if any)?

> WiX Toolset Visual Studio 2015 Extension version 1.0.0.22

* Which version of .NET are you building with?

> .NET Framework 4.6.1

* If the problem occurs when installing your packages built with WiX, what is the version of Windows the package is running on?

> Windows 10

* Describe the problem and the steps to reproduce it.

> Custom Action .NET DLL written to simple Web request over HTTPS fails on attempting to establish connection with TLS 1.2 server. Same program works out of process. I've read how to specify version of .NET Framework using CustomAction.config file (attached).

* Describe the behavior you expected and how it differed from the actual behavior.

> Desire same behavior as when compiling an EXE - observe the correct version of .NET Framework to allow for TLS 1.2 support (or other features of later version of .NET as desired).
[CustomAction.config.txt](https://gith
[CustomAction.config.txt](https://github.com/wixtoolset/issues/files/11425557/CustomAction.config.txt)
ub.com/wixtoolset/issues/files/11425556/CustomAction.config.txt)

**Comment by barnson** [comment]
WiX doesn't use issues for support. We moved this issue to the [GitHub discussions area][gh]. You might also consider asking your question on [Stack Overflow][so].

[gh]: https://github.com/wixtoolset/issues/discussions?discussions_q=category%3AQuestions
[so]: https://stackoverflow.com/questions/tagged/wix

**Comment by BMurri** [comment]
Microsoft [states](https://learn.microsoft.com/dotnet/framework/network-programming/tls) that Framework 4.6 does not support TLS 1..2

> Consider the following recommendations:

> - For TLS 1.2, target .NET Framework 4.7 or later versions on your apps, and target .NET Framework 4.7.1 or later versions on your WCF apps.
> - For TLS 1.3, target .NET Framework 4.8 or later.
> - Do not specify the TLS version. Configure your code to let the OS decide on the TLS version.
> - Perform a thorough code audit to verify you're not specifying a TLS or SSL version.

**Reply by mjb8034** [reply]
I am well aware of what Microsoft states. I have read that article several times and again before posting. The fact is that it works as an EXE, but not as a WiX CA DLL. I can change it to .NET Framework 4.7.1 or .NET Framework 4.8 with same results. The issue is that WiX does not appear to honor the Framework version listed in the documented CustomAction.config file. So, the greater question is how do you get WiX to observe the targeted Framework version of a .NET custom action DLL?

I am not targeting any TLS version. The server is set for TLS 1.2 or 1.3.
Have you tried it yourself?
